src/mesh/grids/non_periodic_cube.cpp

```diff
@@ -0,0 +1,44 @@
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <Sacado.hpp>
+#include "non_periodic_cube.h"
+
+namespace PHiLiP::Grids {
+
+template<int dim, typename TriangulationType>
+void non_periodic_cube(
+    TriangulationType&  grid,
+    double              domain_left,
+    double              domain_right,
+    bool                colorize,
+    const int           left_boundary_id) 
+{
+    dealii::GridGenerator::hyper_cube(grid, domain_left, domain_right, colorize);
+
+    if (left_boundary_id != 9999) {
+        for (auto cell = grid.begin_active(); cell != grid.end(); ++cell) {
+            // Set a dummy material ID
+            cell->set_material_id(9002);
```

Comment:
Not sure that I understand the purpose of the first and last line in this highlighted code

Reply:
The non_periodic_cube_flow is used for `MPI_2D_DUAL_WEIGHTED_RESIDUAL_SSHOCK_H_ADAPTATION` and `MPI_2D_DUAL_WEIGHTED_RESIDUAL_SSHOCK_P_ADAPTATION`. II think for these two cases, the boundary ID is set somewhere else so if I don't bypass the setting of the boundary_id here, those two cases will fail. I could incorporate those two cases into the logic for setting left_boundary_id if that is preferred. 

The material_id line was included because when I was looking through the code for examples how to set boundary_ids, all the code I saw included that line. :sweat_smile: I can try running the tests without that last line and see if it makes a difference.

Reply:
Added a comment explaining the logic of the `left_boundary_id` condition. 
I had to undo the suggested change regarding `set_material_id` as the tests were failing without it.